src/python/py_parsers.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <filesystem>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+#include <pybind11/functional.h>
+
+#include "openvino/genai/parsers.hpp"
+#include "py_utils.hpp"
+#include "openvino/genai/json_container.hpp"
+
+namespace py = pybind11;
+
+using ov::genai::IncrementalParserBase;
+using ov::genai::ParserBase;
+using ov::genai::ReasoningParser;
+using ov::genai::Phi4ReasoningParser;
+using ov::genai::DeepSeekR1ReasoningParser;
+using ov::genai::JsonContainer;
+using ov::genai::Llama32JsonToolParser;
+using ov::genai::Llama32PythonicToolParser;
+using ov::genai::Tokenizer;
+using ov::genai::StreamingStatus;
+
+namespace pyutils = ov::genai::pybind::utils;
+
+namespace {
+
+
+class ConstructableIncrementalParserBase: public IncrementalParserBase {
+public:
+    std::string parse(
+        JsonContainer& msg,
+        const std::string& previous_text, 
+        std::string& delta_text, 
+        const std::optional<std::vector<int64_t>>& previous_tokens = std::nullopt, 
+        const std::optional<std::vector<int64_t>>& delta_tokens = std::nullopt
+    ) override {
+        PYBIND11_OVERRIDE_PURE(
+            std::string,  // Return type
+            IncrementalParserBase,  // Parent class
+            parse,  // Name of function in C++ (must match Python name)
+            msg,
+            previous_text,
+            delta_text,
+            previous_tokens,
+            delta_tokens
+        );
+    }
+};
+
+class ConstructableParserBase: public ParserBase {
+public:
+    void parse(JsonContainer& text) override {
+        PYBIND11_OVERRIDE_PURE(
+            void,  // Return type
+            ParserBase,  // Parent class
+            parse,  // Name of function in C++ (must match Python name)
+            text  // Argument(s)
+        );
+    }
+};
+
+static py::object json_mod = py::module_::import("json");
+
+// wrapper to enhance calling parser from Python
+void call_parser(py::dict& msg, std::function<void(JsonContainer&)> func) {
+    auto msg_anymap = ov::genai::pybind::utils::py_object_to_any_map(msg);
+    auto msg_cpp = JsonContainer(msg_anymap);
```

Comment:
```suggestion
    auto msg_cpp = pyutils::py_object_to_json_container(msg);
```

BTW, conversion through AnyMap does not preserve object keys order as AnyMap sorts them alphabetically